bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/FindReplaceLogicStatus.java

```diff
@@ -0,0 +1,38 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Vector Informatik GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Vector Informatik GmbH - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.ui.internal.findandreplace;
+
+public class FindReplaceLogicStatus implements IFindReplaceLogicStatus {
+
+	public enum MessageCode {
+		NO_MATCH,
+		NONE, WRAPPED, READONLY,
```

Comment:
Where is `NONE` needed? Shouldn't that be an own kind of status object rather than a status code?

Reply:
I agree, especially with the infrastructure in place, it is very easy to add a "NoMessage"-Object, and fits the idiom of our architecture.

Reply:
I have created a new Object that specifically encodes the `NoStatus`-state.

https://github.com/eclipse-platform/eclipse.platform.ui/commit/b8bf9a7e291031b53cf6666060cfd477933e3126